src/NUnitFramework/tests/TestUtilities/TestDirectory.cs

```diff
@@ -42,6 +42,7 @@ public TestDirectory()
             Assume.That(_testDir.Exists, Is.False, _testDir + " should not already exist");
 
             _testDir.Create();
+            _testDir.Refresh();
             Assume.That(_testDir.Exists, Is.True, "Failed to create test dir " + _testDir);
```

Comment:
I no longer recall why this is an assumption and not an assertion. Check that we catch any failure someplace else.